File: app/src/main/java/maderski/bluetoothautoplaymusic/Receivers/PowerReceiver.java
package maderski.bluetoothautoplaymusic.Receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.media.AudioManager;
import android.util.Log;

import maderski.bluetoothautoplaymusic.SharedPrefs.BAPMDataPreferences;
import maderski.bluetoothautoplaymusic.SharedPrefs.BAPMPreferences;
import maderski.bluetoothautoplaymusic.Helpers.PowerHelper;

/**
 * Created by Jason on 7/23/16.
 */
public class PowerReceiver extends BroadcastReceiver {
    private static final String TAG = PowerReceiver.class.getName();

    @Override
    public void onReceive(Context context, Intent intent) {
        //Toast.makeText(context, "BAPM Power Connected", Toast.LENGTH_SHORT).show();

        if(BAPMDataPreferences.getIsSelected(context) && PowerHelper.isPluggedIn(context)) {
            AudioManager audioManager = (AudioManager)context.getSystemService(Context.AUDIO_SERVICE);
            boolean isBTConnected = audioManager.isBluetoothA2dpOn();
            boolean powerRequired = BAPMPreferences.getPowerConnected(context);
            boolean ranActionsOnBtConnect = BAPMDataPreferences.getRanActionsOnBtConnect(context);
            boolean isHeadphones = BAPMDataPreferences.getIsAHeadphonesDevice(context);

            Log.d(TAG, "Power Connected to a Selected BTDevice");
            Log.d(TAG, "Is BT Connected: " + Boolean.toString(isBTConnected));

            if (powerRequired && isBTConnected && !ranActionsOnBtConnect && !isHeadphones) {
                //Toast.makeText(context, "BTAudioPWR Launch", Toast.LENGTH_SHORT).show();
                Intent launchIntent = new Intent();
                launchIntent.setAction("maderski.bluetoothautoplaymusic.pluggedinlaunch");
                context.sendBroadcast(launchIntent);
            }
        }
    }

}

File: app/src/main/java/maderski/bluetoothautoplaymusic/receivers/PowerReceiver.kt
package maderski.bluetoothautoplaymusic.receivers

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log

import maderski.bluetoothautoplaymusic.sharedprefs.BAPMDataPreferences
import maderski.bluetoothautoplaymusic.sharedprefs.BAPMPreferences
import maderski.bluetoothautoplaymusic.utils.BluetoothUtils

/**
 * Created by Jason on 7/23/16.
 */
class PowerReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        val appContext = context.applicationContext
        val isBTConnected = BluetoothUtils.isBluetoothA2DPOnCompat(appContext)
        val powerRequired = BAPMPreferences.getPowerConnected(appContext)
        val ranActionsOnBtConnect = BAPMDataPreferences.getRanActionsOnBtConnect(appContext)
        val isHeadphones = BAPMDataPreferences.getIsAHeadphonesDevice(appContext)

        Log.d(TAG, "Power Connected to a Selected BTDevice")
        Log.d(TAG, "Is BT Connected: " + java.lang.Boolean.toString(isBTConnected))

        if (powerRequired && isBTConnected && ranActionsOnBtConnect.not() && isHeadphones.not()) {
            val launchIntent = Intent()
            launchIntent.action = "maderski.bluetoothautoplaymusic.pluggedinlaunch"
            appContext.sendBroadcast(launchIntent)
        }
    }

    companion object {
        private const val TAG = "PowerReceiver"
    }

}
